README.md

```diff
@@ -1,707 +1,1799 @@
+selectCompletedTodos(store.getState()) // Will not run, and the cached result will be returned.
+store.dispatch(toggleRead(0))
+selectCompletedTodos(store.getState()) // It recalculates.
+```
+
+But why? `selectCompletedTodos` only needs to access `state.todos`, and has nothing to do with `state.alerts`, so why have we broken memoization? Well that's because in `Redux` anytime you make a change to the root `state`, it gets shallowly updated, which means its reference changes, therefore a normal memoization function will always fail the comparison check on the arguments.
+
+But with `Reselect`, we can do something like this:
+
+```ts
+const selectCompletedTodos = createSelector(
+  [(state: RootState) => state.todos],
+  todos => todos.filter(todo => todo.completed === true)
+)
+```
+
+And now we have achieved memoization:
+
+```ts
+selectCompletedTodos(store.getState())
+selectCompletedTodos(store.getState()) // Will not run, and the cached result will be returned.
+store.dispatch(toggleRead(0))
+selectCompletedTodos(store.getState()) // The `input selectors` will run, But the `result function` is skipped and the cached result will be returned.
+```
+
+Even when the overall `state` changes, `Reselect` ensures efficient memoization through its unique approach. The [`result function`] doesn't re-run if the relevant part of the `state` (in this case `state.todos`), remains unchanged. This is due to `Reselect`'s [**_Cascading Double-Layer Memoization_**][**_Cascading Memoization_**]. The first layer checks the entire `state`, and the second layer checks the results of the [`input selectors`]. If the first layer fails (due to a change in the overall `state`) but the second layer succeeds (because `state.todos` is unchanged), `Reselect` skips recalculating the [`result function`]. This dual-check mechanism makes `Reselect` particularly effective in `Redux` applications, ensuring computations are only done when truly necessary.
+
+</details>
+
+<div align="right">[ <a href="#table-of-contents">↑ Back to top ↑</a> ]</div>
+
+---
+
+## API
+
+<a id="createselector"></a>
+
+### createSelector(...inputSelectors | [inputSelectors], resultFunc, createSelectorOptions?)
+
+<details><summary><b>Description</b></summary>
+
+Accepts one or more ["input selectors"][`input selectors`] (either as separate arguments or a single array),
+a single ["result function"][`result function`] / ["combiner"][`combiner`], and an optional options object, and
+generates a memoized selector function.
+
+</details>
+
+<details><summary><b>Parameters</b></summary>
+
+| Name                     | Description                                                                                      |
+| :----------------------- | :----------------------------------------------------------------------------------------------- |
+| `inputSelectors`         | An array of [`input selectors`], can also be passed as separate arguments.                       |
```

Comment:
:hand: We'll to update the terms consistently, but especially here.

Reply:
What "term" are we talking about here?

Reply:
Good question, what _was_ I thinking? :)

It was one of two things:

- We can skip wrapping every single use of one of the "Terminology" terms in links.  Per one of my other comments, we only need to highlight those in the first usage (and we only want to use backticks for actual variable/function names)
- Making sure that whatever final terminology terms we settle on get used consistently, particularly in the API reference section

Reply:
Yes sir, I will fix the backticks and formatting issues.